metals/src/main/scala/scala/meta/internal/builds/BuildTool.scala

```diff
@@ -40,6 +41,12 @@ trait BuildTool {
 
   def executableName: String
 
+  def isBspCompatible(bsp: String): Boolean =
```

Comment:
> It makes me also think that maybe we should check if the file changed is the one for the currently chosen build tool and otherwise suggest to change the build tool used by metals.

Ideally if we have a workflow like adding sbt files, we would show that a new build tool was detected and maybe run the switch command? We should probably do that for each build tools really. 

So we have 1 tool and we modify or add a file for another tool, we should suggest using the new one. This would be much more generic this way.

What do you think?

Reply:
Yes but I don't think it should happen only on create. If you have two build tolls let's say `gradle` and `sbt` and you're using gradle currently in metals but you start making changes in `build.sbt` this should also suggest switching. Currently it will tell you `gradle need to be re-imported` and if you dismiss it it won't show up for changes in `settings.gradle`. I feel like handling multiple build tools should a bit more smooth, so the chances the user thinks that they use a different build tool than the one they use is minimal.

Reply:
Let's try to handle create first and see if it's easy to also react on changes?

Reply:
I won’t solve the problem I’m aiming to solve here… I’m trying do get rid of some of potencially misleading or confusing behaviour.

Reply:
I think the main workflow we are trying to solve is when we have ScalaCLI and then another build tool shows up, I think in that case we should always show an option to switch to that new build tool. If user chooses otherwise we don't show it until user does `switch build server` command

Reply:
This works fine as it is. However if you don’t choose to switch and restart/import build it will switch to bloop + sbt.